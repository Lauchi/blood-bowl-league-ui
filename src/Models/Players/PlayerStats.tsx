export class PlayerStats {
    movement: number;
    strength: number;
    agility: number;
    armor: number;

    constructor(movement: number, strength: number, agility: number, armor: number) {
        this.movement = movement;
        this.strength = strength;
        this.agility = agility;
        this.armor = armor;
    }
}