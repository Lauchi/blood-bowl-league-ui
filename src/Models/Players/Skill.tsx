export class Skill {
    skillId: string;

    constructor(skillId: string) {
        this.skillId = skillId;
    }
}