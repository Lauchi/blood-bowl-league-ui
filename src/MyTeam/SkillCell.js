import React from "react";
import {withTranslation} from "react-i18next";
import {green} from "@material-ui/core/colors";

function SkillCell(props) {
    const { startingSkills, skills, t} = props;
    var startingSkillIds = startingSkills.map(s => s.skillId);
    const additionalSkills = skills.filter(s => startingSkillIds.indexOf(s) < 0);
    const startingSkilltring = startingSkills.map(s => t("skills." + s.skillId)).join(', ');
    const additionalSkillstring = additionalSkills.map(s => t("skills." + s)).join(', ');
    return(
        <>
            <div>
                {startingSkilltring}
            </div>
            <div color={green}>
                {additionalSkillstring}
            </div>
        </>
    )
}

export default withTranslation()(SkillCell)