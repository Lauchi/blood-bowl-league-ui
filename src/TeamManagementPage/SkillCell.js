import React from "react";
import {withTranslation} from "react-i18next";
import {green} from "@material-ui/core/colors";
import Box from "@material-ui/core/Box";

function SkillCell(props) {
    const { startingSkills, skills, t} = props;
    var startingSkillIds = startingSkills.map(s => s.skillId);
    const additionalSkills = skills.filter(s => startingSkillIds.indexOf(s) < 0);
    const startingSkilltring = startingSkills.map(s => t("skills." + s.skillId)).join(', ');
    const additionalSkillstring = additionalSkills.map(s => t("skills." + s)).join(', ');
    return(
        <>
            <Box>
                {startingSkilltring}
            </Box>
            <Box color={green[600]}>
                {additionalSkillstring}
            </Box>
        </>
    )
}

export default withTranslation()(SkillCell)