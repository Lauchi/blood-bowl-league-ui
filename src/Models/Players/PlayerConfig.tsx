import {PlayerStats} from "./PlayerStats";
import {Skill} from "./Skill";

export class PlayerConfig {
    playerStats: PlayerStats;
    startingSkills: Skill[];
    playerTypeId: string;

    constructor(playerStats: PlayerStats, startingSkills: Skill[], playerTypeId: string) {
        this.playerStats = playerStats;
        this.startingSkills = startingSkills;
        this.playerTypeId = playerTypeId;
    }
}