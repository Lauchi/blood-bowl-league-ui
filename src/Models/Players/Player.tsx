import {PlayerConfig} from "./PlayerConfig";

export class Player {
    playerPositionNumber: number;
    playerConfig: PlayerConfig;
    skills: string[];
    starPlayerPoints: number;
    playerTypeId: string;
    playerId: string;


    constructor(playerPositionNumber: number, playerConfig: PlayerConfig, skills: string[], starPlayerPoints: number, playerTypeId: string, playerId: string) {
        this.playerPositionNumber = playerPositionNumber;
        this.playerConfig = playerConfig;
        this.skills = skills;
        this.starPlayerPoints = starPlayerPoints;
        this.playerTypeId = playerTypeId;
        this.playerId = playerId;
    }
}