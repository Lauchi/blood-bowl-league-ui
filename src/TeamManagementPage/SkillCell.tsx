import React, {FunctionComponent} from "react";
import {useTranslation} from "react-i18next";
import {green} from "@material-ui/core/colors";
import Box from "@material-ui/core/Box";
import {Skill} from "../Models/Players/Skill";

const SkillCell:FunctionComponent<{startingSkills: Skill[], skills: string[]}> = ({ startingSkills, skills }) => {
    const { t } = useTranslation();

    const startingSkillIds = startingSkills.map(s => s.skillId);
    const additionalSkills = skills.filter(s => startingSkillIds.indexOf(s) < 0);
    const startingSkillString = startingSkills.map(s => t("skills." + s.skillId)).join(', ');
    const additionalSkillString = additionalSkills.map(s => t("skills." + s)).join(', ');
    return(
        <>
            <Box>
                {startingSkillString}
            </Box>
            <Box color={green[600]}>
                {additionalSkillString}
            </Box>
        </>
    )
}

export default (SkillCell)