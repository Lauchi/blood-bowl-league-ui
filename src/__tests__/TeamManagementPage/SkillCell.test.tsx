import React from 'react';
import renderer from 'react-test-renderer';
import {Skill} from "../../Models/Players/Skill";
import SkillCell from "../../TeamManagementPage/SkillCell";
import {green} from "@material-ui/core/colors";
import {mount} from "enzyme";
import {Box} from "@material-ui/core";

describe('SkillCell', () => {
    const startingSkills = [new Skill("Block")];
    const newSkills = ['Block', 'New_skill1', 'New_skill2'];

    it('snapshot renders', () => {
        const component = renderer.create(<SkillCell skills={newSkills} startingSkills={startingSkills}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('has a the correct color for new skills', () => {
        const wrapper = mount(<SkillCell skills={newSkills} startingSkills={startingSkills}/>);
        const additionalSkillText = wrapper.find(Box).get(1);
        const additionalSkillsColor = additionalSkillText.props.color;
        const innerText = additionalSkillText.props.children;

        expect(additionalSkillsColor).toBe(green[600]);
        expect(innerText).toBe('skills.New_skill1, skills.New_skill2');
    });

    it('has a the correct color for old skills', () => {
        const wrapper = mount(<SkillCell skills={newSkills} startingSkills={startingSkills}/>);
        const oldSkillText = wrapper.find(Box).get(0);
        const oldSkillsColor = oldSkillText.props.color;
        const innerText = oldSkillText.props.children;

        expect(oldSkillsColor).toBeUndefined();
        expect(innerText).toBe('skills.Block');
    });
});