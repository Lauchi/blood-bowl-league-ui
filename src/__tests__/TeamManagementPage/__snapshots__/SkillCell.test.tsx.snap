// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillCell snapshot renders 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    skills.Block
  </div>,
  <div
    className="MuiBox-root MuiBox-root-3"
  >
    skills.New_skill1, skills.New_skill2
  </div>,
]
`;
